pkg/promclient/promclient.go

```diff
@@ -105,7 +105,7 @@ func NewWithTracingClient(logger log.Logger, httpClient *http.Client, userAgent
 
 // req2xx sends a request to the given url.URL. If method is http.MethodPost then
 // the raw query is encoded in the body and the appropriate Content-Type is set.
-func (c *Client) req2xx(ctx context.Context, u *url.URL, method string) (_ []byte, _ int, err error) {
+func (c *Client) req2xx(ctx context.Context, u *url.URL, method string, headers map[string]string) (_ []byte, _ int, err error) {
```

Comment:
Any reason for not reusing the `http.Header` type here instead of `map[string]string`? This definition misses the fact that a header might have multiple values, in fact the best type for headers is `map[string][]string`.

Reply:
Changed to use `http.Header` :+1: